src/views/LandingPage/SignIn/index.vue

```diff
@@ -103,19 +160,92 @@ export default {
       this.$router.push({name: "forgot-password"})
     },
 
-    decryptWallet() {
+    inputMnemonic() {
+      this.$router.push({name: "input-mnemonic"})
+    },
+
+    generateAccount() {
+      this.$router.push({name: "generate-account"})
+    },
+
+    async getKeyStoreLocal() {
       try {
-        this.errorMessage = ""
-        this.pair.unlock(this.password)
+        this.isLoading = true
+        this.dataAccountJson = localStorage.getKeystore()
+        if (this.dataAccountJson === null) {
+          this.isNoAccount = true
+        }
+        
+        this.dataAccount = JSON.parse(this.dataAccountJson)
+
+        const { nonce } = await this.api.query.system.account(
+          this.dataAccount.address
+        )
+
+        this.nonce = nonce
+
+        this.dataMnemonicJson =
+          localStorage.getLocalStorageByName("mnemonic_data")
+        this.dataMnemonic = JSON.parse(this.dataMnemonicJson)
+        this.isLoading = false
+      } catch (err) {
+        console.log(err)
+        this.isloading = false
       }
-      catch (err) {
-        console.error(err)
-        this.snackbar = true
-        this.errorMessage = err.message
-        return
+    },
+
+    setKeystore() {
+      this.keystore = this.dataAccountJson;
+      this.fileName = "keystore";
+    },
+
+    async decryptWallet() {
+      this.isLoading = true
+      this.keystoreInputErrors = []
+      let dataKeystore = []
+
+      const keystore = JSON.parse(this.keystore)
+
+      if (keystore != null) {
```

Comment:
Please change how you apply the logic, too many `if` and `else`. It is hard to maintain this logic.